pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1259,6 +1259,9 @@ public void seek(MessageId messageId) throws PulsarClientException {
 
         long requestId = client.newRequestId();
         MessageIdImpl msgId = (MessageIdImpl) messageId;
+        if (msgId.getEntryId() == -1) {
```

Comment:
This might not be necessary correct. (ledgerId=5, entryId=-1) is a valid position to seek and it should not got to "earliest"

Reply:
Gotcha I will close this PR then